gcp/app-engine/slack_handler.py

```diff
@@ -0,0 +1,271 @@
+from slack_utils import save_and_upload_slack_image, slack_client
+from utils import try_cast_to_int
+
+
+def handle_slack_command(command_text, channel):
+    command_text = command_text.lower()
+
+    msg = ""
+    status_code = 200
+    # Handle tournament info command
+    if "tournament info" in command_text:
+        msg, status_code = handle_tournament_info(command_text, channel)
+
+    # Handle player info command
+    elif "player info image" in command_text:
+        msg, status_code = handle_player_info_image(command_text, channel)
+
+    # Handle player info command
+    elif "player info" in command_text:
+        msg, status_code = handle_player_info(command_text, channel)
+
+    # Handle top 10 players command
+    elif "top 10" in command_text:
+        msg, status_code = handle_top_10_players(command_text, channel)
+
+    # Handle unknown command
+    else:
+        msg = "Sorry, I didn't understand that command. Please try 'tournament info', 'player info image', 'player info', or 'top 10'" + \
+          " (NOTE: for these test endpoints, the only supported player is 'tiger', and the only supported tournament is 'masters'."
+        status_code = 404
+
+    print(f'Finished processing, code={status_code}, msg={msg}')
+    if msg:
+        slack_client.chat_postMessage(channel=channel, text=msg)
+    return msg, status_code
+
+def handle_tournament_info(command_text, channel):
+    # Parse for specific tournament ID or name
+    words = command_text.split()
+    status_code = 200
+    if len(words) > 3:
+        tournament_id = words[-1]  # Assume the last word is the ID
+        tournament = get_next_tournaments(tournament_id)
+        if tournament:
+            response_text = (
+                f"*{tournament['name']}*:\n"
+                f"Location: {tournament['location']}\n"
+                f"Date: {tournament['date']}\n"
+                f"Status: {tournament['status']}\n"
+            )
+        else:
+            response_text = "Tournament not found."
+            status_code = 404
+    else:
+        tournaments = get_next_tournaments()
+        response_text = "*Tournaments:*\n"
+        for tournament in tournaments:
+            response_text += f"- {tournament['name']} ({tournament['status']}) - {tournament['date']}\n"
+
+    if response_text:
+        slack_client.chat_postMessage(channel=channel, text=response_text)
+    return response_text, status_code
+
+def handle_player_info(command_text, channel):
+    # Parse for specific player name/id and tournament name/id
+    words = command_text.split()
+    response_text = None
+    status_code = 200
+    round = None
+    if len(words) > 4:
+        if len(words) > 5:
+            player_search = words[-3]  # Assume third last word is player name/id
+            tournament_search = words[-2]  # Assume second last word is the tournament name/id
+            round = try_cast_to_int(words[-1], None)  # Assume the last word is the round
```

Comment:
also note that for these player info commands, you can optionally specify a round if you want to see a particular round for the player, e.g. `@Golf Better API player info image aaron wyndh 1`